web/src/app/forms/FormContainer.js

```diff
@@ -1,97 +1,64 @@
+import PropTypes from 'prop-types'
 import MountWatcher from '../util/MountWatcher'
 
 import { FormContext, FormContainerContext } from './context'
 import { get, set, cloneDeep } from 'lodash'
 
 // FormContainer handles grouping multiple FormFields.
 // It works with the Form component to handle validation.
-export class FormContainer extends React.PureComponent {
-  static propTypes = {
-    value: p.object,
-
-    errors: p.arrayOf(
-      p.shape({
-        message: p.string.isRequired,
-        field: p.string.isRequired,
-        helpLink: p.string,
-      }),
-    ),
-
-    onChange: p.func,
-    disabled: p.bool,
-
-    mapValue: p.func,
-    mapOnChangeValue: p.func,
-
-    // If true, will render optional fields with `(optional)` appended to the label.
-    // In addition, required fields will not be appended with `*`.
-    optionalLabels: p.bool,
-
-    // Enables functionality to remove an incoming value at it's index from
-    // an array field if the new value is falsey.
-    removeFalseyIdxs: p.bool,
-  }
-
-  static defaultProps = {
-    errors: [],
-    value: {},
-    onChange: () => {},
-
-    mapValue: (value) => value,
-    mapOnChangeValue: (value) => value,
-  }
 
-  state = {
-    validationErrors: [],
-  }
+export function FormContainer(props) {
+  const setState = useRef().current
+  const _fields = useRef().current
+  const { value, mapValue, optionalLabels, disabled: containerDisabled } = props
+  const { disabled: formDisabled, addSubmitCheck } = useContext(FormContext)
 
-  _fields = {}
+  useEffect(() => {
+    addSubmitCheck(onSubmit)
+  }, [])
 
-  addField = (fieldName, validate) => {
-    if (!this._fields[fieldName]) {
-      this._fields[fieldName] = []
+  function addField(fieldName, validate) {
+    if (_fields[fieldName]) {
+      _fields[fieldName] = []
     }
-    this._fields[fieldName].push(validate)
+    _fields[fieldName].push(validate)
 
     return () => {
-      this._fields[fieldName] = this._fields[fieldName].filter(
-        (v) => v !== validate,
-      )
-      if (this._fields[fieldName].length === 0) {
-        delete this._fields[fieldName]
+      _fields[fieldName] = _fields[fieldName].filter((v) => v !== validate)
+      if (_fields[fieldName].length === 0) {
+        delete _fields[fieldName]
       }
     }
   }
 
-  onSubmit = () => {
-    const validate = (field) => {
+  function onSubmit() {
+    function validate(field) {
       let err
       // find first error
-      this._fields[field].find((validate) => {
-        err = validate(get(this.props.value, field))
+      _fields[field].find((validate) => {
+        err = validate(get(props.value, field))
         return err
       })
       if (err) err.field = field
       return err
     }
-    const validationErrors = Object.keys(this._fields)
+    const validationErrors = Object.keys(_fields)
       .map(validate)
       .filter((e) => e)
-    this.setState({ validationErrors })
+    setState({ validationErrors })
```

Comment:
With the above change:
```suggestion
    setPersistentValidationErrors(validationErrors)
```